components/log4j/org.wso2.carbon.analytics.shared.data.agents.log4j/src/main/java/org/wso2/carbon/analytics/shared/data/agents/log4j/appender/LogEventAppender.java

```diff
@@ -415,29 +373,28 @@ private void publishLogEvent(TenantDomainAwareLoggingEvent event) throws ParseEx
             String priority = priorityLayout.format(event);
             String message = messageLayout.format(event);
             String ip = ipLayout.format(event);
-            String instance = (getInstanceId() == null || getInstanceId().isEmpty()) ? instanceLayout.format(event) : getInstanceId();
+            String instance = (getInstanceId() == null || getInstanceId().isEmpty()) ? instanceLayout.format(event) :
+                    getInstanceId();
             String stacktrace = "";
 
             if (isStackTrace) {
                 if (event.getThrowableInformation() != null) {
                     stacktrace = getStacktrace(event.getThrowableInformation().getThrowable());
-                } else {
-                    stacktrace = "";
                 }
             }
             DateFormat formatter = new SimpleDateFormat(LoggingConstants.DATE_TIME_FORMATTER);
             Date date = formatter.parse(logTime);
             Map<String, String> arbitraryDataMap = new HashMap<String, String>();
-            arbitraryDataMap.put(columns[0], serverName);
-            arbitraryDataMap.put(columns[1], appName);
-            arbitraryDataMap.put(columns[2], String.valueOf(date.getTime()));
-            arbitraryDataMap.put(columns[3], logger);
-            arbitraryDataMap.put(columns[4], priority);
-            arbitraryDataMap.put(columns[5], message);
-            arbitraryDataMap.put(columns[6], ip);
-            arbitraryDataMap.put(columns[7], instance);
+            arbitraryDataMap.put(COLUMNS[0], serverName);
+            arbitraryDataMap.put(COLUMNS[1], appName);
+            arbitraryDataMap.put(COLUMNS[2], String.valueOf(date.getTime()));
+            arbitraryDataMap.put(COLUMNS[3], logger);
+            arbitraryDataMap.put(COLUMNS[4], priority);
+            arbitraryDataMap.put(COLUMNS[5], message);
+            arbitraryDataMap.put(COLUMNS[6], ip);
+            arbitraryDataMap.put(COLUMNS[7], instance);
```

Comment:
This is fixed in following commit : https://github.com/wso2/shared-analytics/pull/140/commits/7fee3a3d4b418a3f172d7e6ca5d5849d5c4a708b